### PR TITLE
Switch to pyproject.toml and ruff

### LINKED ISSUE #219: Use ruff
Would you be open for a PR switching all the lint and formatting tools to use [ruff](https://github.com/astral-sh/ruff) instead?

It does what `flake8`, `pylint`, `isort` and `black` does in a single tool and it is a lot faster. After getting used to it I often get a bit annoyed at how slow `pylint` is.

### PR DESCRIPTION
**Pull request by DeviousStoat**
fixes #219

One thing to note, with `pyproject` the `importlib.metadata` doesn't have the `Author` key so I had to change to use `Author-email` for the copyright in the docs.
So your name looks like `Derrick Gilland <dgilland@gmail.com>` instead of just `Derrick Gilland` in the copyright in the documentation page.
Seems to be a [known issue](https://stackoverflow.com/questions/75801738/importlib-metadata-doesnt-appear-to-handle-the-authors-field-from-a-pyproject-t). I can change it if you want but it didn't sound too bad.

### CODE REVIEW

File: tasks.py [hunk 1]
```diff
@@ -32,66 +34,46 @@
...
-
-
-@task()
-def docformatter(ctx: Context, target: t.Optional[str] = None) -> None:
-    """Autoformat docstrings using docformatter."""
-    run(
-        f"docformatter -r {target or LINT_TARGETS} "
-        f"--in-place --pre-summary-newline --wrap-descriptions 100 --wrap-summaries 100"
-    )
+def ruff_lint_fix(ctx: Context, target: str, quiet: bool = False) -> None:
```
Comment (dgilland) [inline tasks.py:43]:
```suggestion
def ruff_fix(ctx: Context, target: str, quiet: bool = False) -> None:
```

File: tasks.py [hunk 2]
```diff
@@ -32,66 +34,46 @@
...
+    run(f"mypy {LINT_TARGETS} --no-error-summary")
 
 
 @task()
 def lint(ctx: Context) -> None:
     """Run linters."""
     linters = {
-        "flake8": flake8,
-        "pylint": pylint,
+        "ruff_lint": ruff_lint,
```
Comment (dgilland) [inline tasks.py:76]:
```suggestion
        "ruff-lint": ruff_lint,
```

File: tests/test_functions.py [hunk 3]
```diff
@@ -178,7 +178,7 @@ def func():
     "func,wait,args,kwargs,expected",
     [(lambda a, b, c: (a, b, c), 250, (1, 2), {"c": 3}, (1, 2, 3))],
 )
-def test_delay(mock_sleep, func, wait, args, kwargs, expected):
+def test_delay(mock_sleep, func, wait, args, kwargs, expected):  # noqa: PLR0913
```
Comment (dgilland) [inline tests/test_functions.py:181]: Thinking to disable `PLR0913`.

**Review by dgilland — CHANGES_REQUESTED** [review]
Overall, looks good! Just had a few nits.

> One thing to note, with pyproject the importlib.metadata doesn't have the Author key so I had to change to use Author-email for the copyright in the docs.

That's fine.

**Comment by dgilland** [comment]
Really appreciate the work here!

File: tasks.py [hunk 4]
```diff
@@ -15,15 +15,17 @@
 import tempfile
 import typing as t
 
-from invoke import Context, Exit, UnexpectedExit, run as _run, task
+from invoke import run as _run
+from invoke.context import Context
+from invoke.exceptions import Exit, UnexpectedExit
+from invoke.tasks import task
```
[the remark this answers is not in the document]
Comment (DeviousStoat) [reply tasks.py:21]: Oh this was just me manually changing that, because the module doesn't export these. But I can change it back, it's probably safe

**Comment by DeviousStoat** [comment]
Applied all the requested changes

**Pull request merged**